Sources/Sentry/SentryHttpTransport.m

```diff
@@ -76,16 +80,36 @@ - (id)initWithOptions:(SentryOptions *)options
         [self.fileManager setDelegate:self];
 
         [self sendAllCachedEnvelopes];
+
+#if !TARGET_OS_WATCH
+        [SentryReachability
+               monitorURL:[NSURL URLWithString:@"https://sentry.io"]
+            usingCallback:^(BOOL connected, NSString *_Nonnull typeDescription) {
```

Comment:
`h`: I think we should add tests for this new type of logic. We could pass in SentryReachability as a dependency, subclass in it tests, and replace it with a test class we control in the tests.